tests/unit/query_graph/sql/adapter/test_snowflake.py

```diff
@@ -0,0 +1,93 @@
+"""
+Test snowflake adapter module
+"""
+import textwrap
+
+from featurebyte.query_graph.sql.adapter import SnowflakeAdapter
+from tests.unit.query_graph.sql.adapter.base_adapter_test import BaseAdapterTest
+
+
+class TestSnowflakeAdapter(BaseAdapterTest):
+    """
+    Test snowflake adapter class
+    """
+
+    adapter = SnowflakeAdapter()
+
+    @classmethod
+    def get_group_by_expected_result(cls) -> str:
+        """
+        Returns expected result of group by query
+        """
+        return textwrap.dedent(
+            """
+                SELECT
+                  VECTOR_T0."serving_name" AS "serving_name",
+                  VECTOR_T0."serving_name_2" AS "serving_name_2",
+                  GROUP_BY_RESULT."sum_result" AS "sum_result",
+                  GROUP_BY_RESULT."avg_result" AS "avg_result",
+                  VECTOR_T0."result" AS "result",
+                  VECTOR_T1."result2" AS "result2",
+                  VECTOR_T2."result3" AS "result3"
+                FROM (
+                  SELECT
+                    INITIAL_DATA."serving_name" AS "serving_name",
+                    INITIAL_DATA."serving_name_2" AS "serving_name_2",
+                    AGG_0."VECTOR_AGG_RESULT" AS "result"
+                  FROM (
+                    SELECT
+                      REQ."serving_name" AS "serving_name",
+                      REQ."serving_name_2" AS "serving_name_2",
+                      TABLE."parent" AS "parent"
+                  ) AS INITIAL_DATA, TABLE(
```

Comment:
Does this mean the `INITIAL_DATA` query will be repeated in every `VECTOR_T{}` subqueries? Probably fine for now, but I wonder if this will cause performance issue when the initial data is complex such as ItemView + SCD joins.

Reply:
yea that's probably the case. i'll add a ticket to see if we can optimize this at a later point once i finish the tiling version as well

Reply:
https://featurebyte.atlassian.net/browse/DEV-2179